src/content/components/text-input/usage.mdx

```diff
@@ -15,11 +30,39 @@ _Text inputs_ enable the user to interact with and input data. Use when the appl
 
 ## Format
 
+### Default values
+
+Where possible, add programmatic assistance. Detect and pre-fill inputs to reduce errors and save time. When the software can't determine the value that belongs in an input, use type-ahead to make suggestions. Use sentence-case for default values, detected values, and auto-completion text.
+
+### Validation and errors
+
+Real time validation helps to streamline the process and keep data clean when the user is filling out forms. For full guidelines, refer to the [forms](/components/form/usage) usage page.
+
+<ImageComponent cols="8" caption="Error example">
+
+![Validation text example](images/text-input-usage-2.png)
+
+</ImageComponent>
+
+### Character count
+
+Text fields and text areas may have a character limit and counter. Use the character limit and counter to let users know how long their entry can be. 
```

Comment:
> to let users know how long their entry can be **a head of time** 

I think the "a head of time" is important to keep.

Reply:
It's in!